presto-main/src/main/java/com/facebook/presto/sessionpropertyproviders/JavaWorkerSystemSessionPropertyProvider.java

```diff
@@ -0,0 +1,223 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sessionpropertyproviders;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.spi.session.PropertyMetadata;
+import com.facebook.presto.spi.session.SystemSessionPropertyProvider;
+import com.facebook.presto.spiller.NodeSpillConfig;
+import com.facebook.presto.sql.analyzer.JavaFeaturesConfig;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+
+import java.util.List;
+
+import static com.facebook.presto.SystemSessionProperties.isSpillEnabled;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.facebook.presto.spi.session.PropertyMetadata.booleanProperty;
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class JavaWorkerSystemSessionPropertyProvider
+        implements SystemSessionPropertyProvider
+{
+    public static final String AGGREGATION_SPILL_ENABLED = "aggregation_spill_enabled";
+    public static final String TOPN_SPILL_ENABLED = "topn_spill_enabled";
+    public static final String DISTINCT_AGGREGATION_SPILL_ENABLED = "distinct_aggregation_spill_enabled";
+    public static final String DEDUP_BASED_DISTINCT_AGGREGATION_SPILL_ENABLED = "dedup_based_distinct_aggregation_spill_enabled";
+    public static final String DISTINCT_AGGREGATION_LARGE_BLOCK_SPILL_ENABLED = "distinct_aggregation_large_block_spill_enabled";
+    public static final String DISTINCT_AGGREGATION_LARGE_BLOCK_SIZE_THRESHOLD = "distinct_aggregation_large_block_size_threshold";
+    public static final String ORDER_BY_AGGREGATION_SPILL_ENABLED = "order_by_aggregation_spill_enabled";
+    public static final String WINDOW_SPILL_ENABLED = "window_spill_enabled";
+    public static final String ORDER_BY_SPILL_ENABLED = "order_by_spill_enabled";
+    public static final String AGGREGATION_OPERATOR_UNSPILL_MEMORY_LIMIT = "aggregation_operator_unspill_memory_limit";
+    public static final String TOPN_OPERATOR_UNSPILL_MEMORY_LIMIT = "topn_operator_unspill_memory_limit";
+    public static final String TEMP_STORAGE_SPILLER_BUFFER_SIZE = "temp_storage_spiller_buffer_size";
+    public static final String OFFSET_CLAUSE_ENABLED = "offset_clause_enabled";
+    public static final String TABLE_WRITER_MERGE_OPERATOR_ENABLED = "table_writer_merge_operator_enabled";
+
+    private final List<PropertyMetadata<?>> sessionProperties;
+
+    public JavaWorkerSystemSessionPropertyProvider(JavaFeaturesConfig javaFeaturesConfig, NodeSpillConfig nodeSpillConfig)
```

Comment:
As a follow-on to my previous comment, let's `@Inject` the dependencies here and statically define it in Guice.